core/src/main/scala/kafka/common/ZkNodeChangeNotificationListener.scala

```diff
@@ -58,6 +59,7 @@ class ZkNodeChangeNotificationListener(private val zkUtils: ZkUtils,
   def init() {
     zkUtils.makeSurePersistentPathExists(seqNodeRoot)
     zkUtils.zkClient.subscribeChildChanges(seqNodeRoot, NodeChangeListener)
+    zkUtils.zkClient.subscribeStateChanges(ZkStateChangeListener)
     processAllNotifications()
```

Comment:
I mention the child changed state because we register for child changes as well, so processAllNotifications() should be called on child changes. 

You are right, this may have duplicate zookeeper calls but the processing of notification is idempotent and needs to be given the notification can not be deleted until the expiration time is reached  , this means if a broker processes a notification -> goes down -> comes back up , it will still see the notifications again and it is suppose to handle it correctly and we do that by ensuring the process notification method is idempotent.